### ISSUE TITLE
Reposense not detecting my commits after 21st October

### ISSUE THREAD

**Issue by ooihuiying** [nus-cs2103-AY1819S1/forum#126]
Hi, i am facing this issue where the commits made by me after 21st October are not detected by reposense. My code has been merged into the master branch of my team repo but somehow when i click on the rightmost Ramp(triangle) for my account on the dashboard, it shows that my latest commit was made on the 21st October. Thank you for the help in advance.

**Comment by ghost** [comment]
Hi, did you check against the updated ` _reposense/config.json` file template which can be found [here](https://nus-cs2103-ay1819s1.github.io/cs2103-website/admin/reposenseConfigTemplates.html).

There was a mistake in the JSON file earlier. I think missing a comma after each `AuthorNames`.

Good luck & Have Fun! :)

**Comment by damithc** [comment]
@ooihuiying can you post the link to your repo's _reposense/config.json?

**Comment by ooihuiying** [comment]
https://github.com/ooihuiying/main/blob/0a5d48b1dc1bd236dd504af53025e2bf2de1e7c6/_reposense/config.json

Is this the link you are looking for? The commas look correct.

Thank you.

**Comment by ghost** [comment]
We (OP and myself) corresponded via email and I suggested to her 2 other possibilities (not exhaustive):

1.  Her email address is not configured properly with her GitHub account.
2. _Locally,_ she did not set her Git Username to match her GitHub Username. Refer to [here](https://nus-cs2103-ay1819s1.github.io/cs2103-website/admin/appendixE-gitHub.html#setting-git-username-to-match-github-username)

![image](https://user-images.githubusercontent.com/35895966/47611241-66d83980-da9c-11e8-9ef0-cc209769d674.png)

The case seems suspect especially if her other teammates do not face the issue she does.

Hope this helps!

**Comment by damithc** [comment]
Thanks for the help @HEARTOFAL1ON 
Possibility 1 is unlikely to affect RepoSense but 2 can. If commits have multiple Git author names, the config file should be updated accordingly. In the meantime,  I have asked the RepoSense team to investigate.

**Comment by ghost** [comment]
No prob!

**Comment by eugenepeh** [comment]
@ooihuiying 

Hi huiying, 

Note that your commits on 25th was only merged on the 27th, hence the module's dashboard which was generated on the 26th does not contain your commits.

Also, as of now RepoSense does not include merge commit into the ramp chart as it generally doesn't include any code addition. Hence your merge commits on the 23rd and 24th were also excluded from the chart. 

Hope that answers your concern!

Best regards,
Eugene
On behalf of RepoSense development team

![image](https://user-images.githubusercontent.com/19277206/47624174-8468da00-db54-11e8-8206-dfb4a77abe6e.png)
![image](https://user-images.githubusercontent.com/19277206/47624184-a82c2000-db54-11e8-9628-c60ceddde750.png)

![image](https://user-images.githubusercontent.com/19277206/47624179-9f3b4e80-db54-11e8-9d9b-87c47dab0aea.png)

**Comment by ooihuiying** [comment]
I see! Sorry for the trouble, and thank you so much!

**Comment by eugenepeh** [comment]
>I see! Sorry for the trouble, and thank you so much!

No problem, you're welcome. We welcome feedback to help us better improve the product 👍

**Comment by damithc** [comment]
I have updated the dashboard. @ooihuiying see if the missing commits are reflected now.

**Comment by jjlee050** [comment]
Can i check if the reposense is checking based on our team repo main branch? This is because the reposense does not detect some of my commits. One instance is between 24 October to 26th October.

**Comment by eugenepeh** [comment]
>Can i check if the reposense is checking based on our team repo main branch? This is because the reposense does not detect some of my commits. One instance is between 24 October to 26th October.

Yes. As highlighted above, _merge commits_ are omitted from the ramp chart as it generally doesn't include any code addition.

**Comment by jjlee050** [comment]
![screenshot 2018-10-29 at 18 08 18](https://user-images.githubusercontent.com/5901764/47643182-a2125f80-dba5-11e8-916c-17fd998aff25.png)
@eugenepeh However, on the oct 26, that commit is not a merge commit however it is not included in reposense. 
![screenshot 2018-10-29 at 18 13 43](https://user-images.githubusercontent.com/5901764/47643416-675cf700-dba6-11e8-8527-936bbd30daf8.png)
Same for 21 and 22 October. This is not reflected in reposense.

**Comment by ooihuiying** [comment]
@damithc 

Yes, it is reflected correctly. Thank you.

**Comment by eugenepeh** [comment]
@jjlee050 

Your team need to include all the author names
E.g. for your case, you will need to key in `Joseph Lee Zhuo Xun` as your author name in your team's config as well.

![image](https://user-images.githubusercontent.com/19277206/47645482-25cf4a80-dbac-11e8-8e17-b1a2e9eb4d4a.png)

Hope it helps!

Best regards,
Eugene

**Comment by jjlee050** [comment]
Thanks @eugenepeh . Will edit my reposense json.

**Comment by pikulet** [comment]
Can we opt out of RepoSense? My team has been using squash and merge :(

We will use collate.

**Comment by eugenepeh** [comment]
>Can we opt out of RepoSense? My team has been using squash and merge :(

RepoSense works with squash and merge, it doesn't pose any issue.
In fact if you marked your code as suppose to what you do for collate, RepoSense will work the same way. 
So there isn't actually any reason to go back to collate.

**Comment by yong24s** [comment]
@pikulet

As your team did not configure your copy of `_reposense/config.json`, RepoSense used the suggested template from `https://nus-cs2103-ay1819s1.github.io/cs2103-website/admin/reposenseConfigTemplates.html`, which assumes that `github id` and `author name` are the same.

```
$ git shortlog -e -s -n | grep -i jy****@*****.com.sg
    28  Joyce <jy****@*****.com.sg>
    1  pikulet <jy****@*****.com.sg>
```

From the above command, you have commits under two different author names.

```
    {
      "githubId": "pikulet",
      "displayName": "JOYCE...HUHUI",
      "authorNames": ["pikulet", "Joyce"]
    }
```	

So you need to add the another author name `Joyce` to `_reposense/config.json` as well.

Follow the template given in the above `link`, you only need to add author names for your's and Adam's.

Hope it helps!

Best Regards,
Yong Hao

**Comment by pikulet** [comment]
Hi Yong Hao and eugene, that was very helpful. Thank you very much! :)

**Comment by A19Sean** [comment]
![image](https://user-images.githubusercontent.com/37375137/47861368-9d5fdc80-de2d-11e8-9861-8966466a0b80.png)

Reposense isn't detecting any contributions from me for v1.3, but works fine for my other teammates. We did not configure our reposense.json file. May I ask what may be the issue here?

**Comment by yong24s** [comment]
@A19Sean 

```
$ git shortlog -e -s | grep sean.*****@*****.com
    35  A19Sean <sean.*****@*****com>
    29  Auyok Sean <sean.*****@*****.com>
```
From above, you probably either work with multiple computers or changed your git username.

For the unaccounted 29 commits, you should really configure `_reposense/config.json`.

Read https://github.com/nus-cs2103-AY1819S1/forum/issues/126#issuecomment-433937312 for instructions.

Yong Hao

**Comment by A19Sean** [comment]
@yong24s I managed to solve the issue, thanks a lot for your help!
